src/mlpack/core/data/tokenizers/char_extract.hpp

```diff
@@ -36,15 +36,20 @@ class CharExtract
    *
    * @param str String view to retrieve the next token from.
    */
-  int operator()(boost::string_view& str) const
+  int operator()(boost::string_view& str, int ngram = 1) const
   {
+    int retval = 0;
     if (str.empty())
       return EOF;
-
-    const int retval = static_cast<unsigned char>(str[0]);
-
+    int count = 0;
+    while(ngram--)
+    {
+      retval = retval * pow(10,count) + (static_cast<unsigned char>(str[count]));
+      count++;
+      if(count>=str.length())
+        break;
+    }
     str.remove_prefix(1);
```

Comment:
We need to find a better formula to differentiate between `abc` and `cba` when ngram is 3, taking 3 at a time